src/KafkaFlow.Admin/TelemetryScheduler.cs

```diff
@@ -56,29 +57,45 @@ public void Stop(string telemetryId)
             IList<IMessageConsumer> consumers,
             IMessageProducer producer)
         {
-            var items = consumers.SelectMany(
-                c => c.Assignment.Select(
-                    a => new ConsumerTelemetryMetric()
+            var items = consumers
+                .Where(c => c.Subscription?.Any() == true)
+                .SelectMany(c =>
+                    c.Subscription?.Select(topic => new ConsumerTelemetryMetric
                     {
                         ConsumerName = c.ConsumerName,
-                        Topic = a.Topic,
+                        Topic = topic,
                         GroupId = c.GroupId,
                         InstanceName = Environment.MachineName,
                         PausedPartitions = c.PausedPartitions
-                            .Where(p => p.Topic == a.Topic)
+                            .Where(p => p.Topic == topic)
                             .Select(p => p.Partition.Value),
                         RunningPartitions = c.RunningPartitions
-                            .Where(p => p.Topic == a.Topic)
+                            .Where(p => p.Topic == topic)
                             .Select(p => p.Partition.Value),
                         WorkersCount = c.WorkersCount,
                         Status = c.Status,
-                        SentAt = DateTime.Now,
+                        Lag = c.Assignment
+                            .Where(a => a.Topic == topic)
+                            .Sum(tp => LatestOffset(c, tp) - LastCommittedOffset(c, tp)),
+                        SentAt = DateTime.Now.ToUniversalTime(),
                     }));
 
             foreach (var item in items)
             {
                 producer.Produce(topicName, Guid.NewGuid().ToByteArray(), item);
             }
         }
+
+        private static long LastCommittedOffset(IMessageConsumer c, TopicPartition tp)
+        {
+            var offset = c.GetPosition(tp);
```

Comment:
I don´t know if is a good idea to call the GetPosition() inside a LINQ and repeated this LINQ execution every 1s.

Reply:
right, please take a look again :)